profile.md

```diff
@@ -472,10 +472,21 @@ Examples values of the `scope` claim:
 
 ### Group Based Authorization: wlcg.groups 
 
-Authorization may be based on the `wlcg.groups` claim. The value of the `wlcg.groups` claim is an ordered JSON array of case-sensitive strings reflecting the VO groups of which the token subject is a member. 
+Authorization may be based on the `wlcg.groups` claim. The value of the `wlcg.groups` claim is an ordered JSON array of case-sensitive strings reflecting the asserted group membership of the subject of the token.
 
-`wlcg.groups `semantics are equivalent to existing VOMS groups. VOMS roles should be considered as optional (i.e. returned only if requested) `wlcg.groups`; selection of optional groups is discussed below.
+For authorization purposes, group membership is considered hierarchical.  That is, membership in a child group (`/cms/uscms`) implies membership in the parent group.  For authorization, calculated implied group memberships should be added to the end of the group list.  That is, the following list of groups:
```

Comment:
"should be added" --> "MUST be added."

Reply:
Also, I would drop the word `implies` in `membership in a child group ... implies membership in the parent group.`  To me, using implies suggests the parent may be omitted.  This could be true, but it conflates two issues: the hierarchy and that hierarchy's representation.

To me, a better way of stating this would be to use a statement of fact; i.e., the verb "to be".  For example:

    A member of a group is always a member of all parents of that group.

To be more complete (with a slightly different defn of "parent"), I would suggest:

  * Groups are hierarchical through a parent relationship; that is, each group has optionally a single parent group.
  * Groups without a parent group are top-level groups.
  * Groups with a parent group are child groups.
  * No group can be both the parent of some other group and have that other group its parent (no loops).
  * Anyone who is a member of a child group is always a member of that group's parent groups.

Note that the last definition is recursive, always terminating at some top-level group.

This describes the hierarchy concept without saying which of these groups the OP is required to assert.

Reply:
@paulmillar -- yes, I struggled with that as well.  I like your approach.  Could you do a PR against this branch?

Reply:
I think we haven't yet reached consensus within the WG, so I suggest we hold off until the discussion reaches a conclusion.